src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6390,9 +6395,8 @@ synchronized final boolean readPacket() throws SQLServerException {
             // terminate the connection.
             if ((command.getCancelQueryTimeoutSeconds() > 0 && command.getQueryTimeoutSeconds() > 0)) {
```

Comment:
I know this wasn't edited but shouldn't this be a `||` and not `&&`.

Reply:
I was curious about that as well but didn't touch it as wasn't entirely sure where the second half of that is loaded from.

Seems like it should be if either is defined with values less than zero considered to be zero:

```java
int seconds = Math.ceil(command.getCancelQueryTimeoutSeconds(),0)
     + Math.ceil(command.getQueryTimeoutSeconds(),0);
if (seconds > 0) {
    // if a timeout is configured with this object, add it to the timeout poller
    this.timeout = con.getSharedTimer().schedule(new TdsTimeoutTask(command, con), seconds);
}
```

Reply:
I think I figured it out, the intention might've been
`(command.getCancelQueryTimeoutSeconds() > -1 && command.getQueryTimeoutSeconds() > 0)`

cancelQueryTimeout's "infinite" representation is -1, with 0 meaning literally 0. Whereas queryTimeout has an infinite timeout representation value of both 0 and -1. So this is meant to not schedule a timeout when either of the values are infinite.

Reply:
I just read through that link three times and still have no clue what "cancelQueryTimeout" is supposed to do:

> __cancelQueryTimeout__  Beginning with Microsoft JDBC Driver 6.4 for SQL Server, this property can be used to cancel queryTimeout set on the connection. Query execution hangs and does not throw exception if TCP connection to SQL Server is silently dropped. This property is only applicable if 'queryTimeout' is also set on the connection.
> The driver waits the total amount of cancelQueryTimeout + queryTimeout seconds, to drop the connection and close the channel. 

I can't tell if the space in "cancel queryTimeout" is a typo or if this property is meant to *cancel* a `queryTimeout` setting.

Reply:
@sehrope Yes this property is used to cancel 'QueryTimeout' and is only functional if queryTimeout is provided (not infinite) - hence the && check. This was introduced to fix #525 .